tkg/client/delete_region.go

```diff
@@ -446,15 +453,23 @@ func (c *TkgClient) verifyProviderConfigVariablesExists(providerName string) err
 	return errors.Errorf("value for variables [%s] is not set. Please set the value using os environment variables or the tkg config file", strings.Join(missingVariables, ","))
 }
 
-func (c *TkgClient) cleanUpAVIResourcesInManagementCluster(regionalClusterClient, cleanupClusterClient clusterclient.Client, clusterName, clusterNamespace string) error {
+func (c *TkgClient) cleanUpAVIResourcesInManagementCluster(regionalClusterClient clusterclient.Client, isFailure bool, clusterName, clusterNamespace string) error {
+	var akoAddonSecretName string
 	akoAddonSecret := &corev1.Secret{}
 
 	// Ako addon secret name is different between legacy cluster and classy cluster
-	akoAddonSecretName := constants.AkoAddonName + "-data-values"
-	isClusterClassBased, err := cleanupClusterClient.IsClusterClassBased(clusterName, clusterNamespace)
+	// Ako addon secret name depends on if management cluster is successfully created
```

Comment:
This can be simplified as:

```go
akoAddonSecretName := constants.AkoAddonName + "-data-values"
if isFailure {
	akoAddonSecretName = constants.AkoAddonName + "-addon"
}
```